File: CVAndVacancyBase.Tests/Controllers/HomeControllerTest.vb
Imports System.Web.Mvc
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports CVAndVacancyBase

<TestClass()> Public Class HomeControllerTest
    <TestMethod()> Public Sub Index()
        'Упорядочение
        Dim controller As New HomeController()

        'Действие
        Dim result As ViewResult = DirectCast(controller.Index(), ViewResult)

        'Утверждение
        Assert.IsNotNull(result)
        Dim viewData As ViewDataDictionary = result.ViewData
        Assert.AreEqual("Home Page", viewData("Title"))
    End Sub
End Class

File: CVAndVacancyBase.Tests/Controllers/HomeControllerTest.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using CVAndVacancyBase;
using CVAndVacancyBase.Controllers;

namespace CVAndVacancyBase.Tests.Controllers
{
    [TestClass]
    public class HomeControllerTest
    {
        [TestMethod]
        public void Index()
        {
            // Упорядочение
            HomeController controller = new HomeController();

            // Действие
            ViewResult result = controller.Index() as ViewResult;

            // Утверждение
            Assert.IsNotNull(result);
            Assert.AreEqual("Home Page", result.ViewBag.Title);
        }
    }
}
